src/math/Vector2.d.ts

```diff
@@ -427,6 +429,7 @@ export class Vector2 implements Vector {
 	 * @return The created or provided array.
 	 */
 	toArray( array?: number[], offset?: number ): number[];
+	toArray( array?: Vector2tuple, offset?: 0 ): Vector2tuple;
```

Comment:
I believe this valid usage will cause a compilation error...

```js
const tmp = [];
vector.toArray(tmp);
```

... because the argument is not of type Vector2tuple.

Reply:
Thanks for providing the test case.  It does compile for me.  If I right-click on your call to toArray in Visual Studio Code and choose "Go to definition..." from the context menu, it takes me to the existing definition of toArray, not the special case that I have inserted.  The compiler seems to be able to match the correct definition as required.  I imagine you'll want to play with it yourself to satisfy yourself of this and try and find some means of breaking the definition that I have provided.